corehq/apps/userreports/sql/adapter.py

```diff
@@ -216,6 +228,24 @@ def doc_exists(self, doc):
             query = session.query(self.get_table()).filter_by(doc_id=doc['_id'])
             return session.query(query.exists()).scalar()
 
+    def get_rows(self, doc_id):
+        """Returns a list of entries from the datasource matching the `doc_id`. Each entry is a dictionary with
+        the column name as the key and value as the value
+        """
+        table = self.get_table()
+        headers = list(table.columns.keys())
+
+        named_entries = []
+        with self.session_context() as session:
+            query = session.query(self.get_table()).filter_by(doc_id=doc_id)
```

Comment:
nit: ideally there should only be one row right? should we raise an error if there is more than 1?

Reply:
There are cases where a document can result in multiple rows.

Reply:
Can you say more?
I would think one form/case would have only one row in the table.

Reply:
I thought so too, but [there are ways to iterate form/case data](https://commcare-hq.readthedocs.io/ucr.html#saving-multiple-rows-per-case-form).

Reply:
We've confirmed that since we only fetch the rows when the repeat record is sent, and the payload structure supports multiple rows for a doc_id, there is no problem with this approach.